Mathlib/Data/ZMod/Topology.lean

```diff
@@ -0,0 +1,24 @@
+/-
+Copyright (c) 2023 Ashvni Narayanan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Ashvni Narayanan
+-/
+import Mathlib.Data.ZMod.Basic
+import Mathlib.Topology.Algebra.Constructions
+
+/-!
+This file adds a discrete topological structure to `ZMod n` for all `n`.
+-/
+
+namespace ZMod
+/-- Making `ZMod` a discrete topological space. -/
+instance {d : ℕ} : TopologicalSpace (ZMod d) := ⊥
+
+instance {d : ℕ} : DiscreteTopology (ZMod d) := { eq_bot := rfl }
+end ZMod
+
+@[continuity]
+lemma induced_top_cont_inv {n : ℕ} : @Continuous _ _ (TopologicalSpace.induced
+    (Units.coeHom (ZMod n)) inferInstance) _ (Units.inv : (ZMod n)ˣ → ZMod n) := by
```

Comment:
We have a canonical topology on the units. Why not use it?

Reply:
Can you not use the universal property of the induced topology to state that `embedProduct` is the induced topology without actually fiddling with several topologies on the same type?

Reply:
You should add this instance (if we don't have it yet):
```
@[to_additive]
instance {M : Type*} [Monoid M] [TopologicalSpace M] [DiscreteTopology M] :
    DiscreteTopology Mˣ :=
  .of_continuous_injective Units.continuous_val Units.ext
```

Reply:
BTW, you can also have `Embedding Units.val` from `Units.embedding_val_mk`.

Reply:
> Can you not use the universal property of the induced topology to state that `embedProduct` is the induced topology without actually fiddling with several topologies on the same type?

These are equivalent conditions, as shown in the lemma `top_eq_iff_cont_inv`